skyportal/onboarding.py

```diff
@@ -18,68 +23,88 @@
 
 def create_user(strategy, details, backend, uid, user=None, *args, **kwargs):
     invite_token = strategy.session_get("invite_token")
-
-    existing_user = DBSession().query(User).filter(User.oauth_uid == uid).first()
-
-    if cfg["invitations.enabled"]:
-
-        if existing_user is None and invite_token is None:
-            raise Exception(
-                "Authentication Error: Missing invite token. A valid invite token is required."
-            )
-        elif existing_user is not None:
-            return {"is_new": False, "user": existing_user}
-
+    with DBSession() as session:
         try:
```

Comment:
I think that if you use `with DBSession() as session:` we don't need to rollback manually in case of an exception.

Reply:
Noted. I'll remove those then!